File: TypeScript/0-introduccion.ts
//Imprimir un mensaje:
console.log("Hola, mundo!");

//Variables con tipos:
let edad1: number = 30;
let estatura: number = 1.75;
let genero: string = 'M';
let casado: boolean = false;
let nombre: string = "Juan";
console.log("Mi nombre es " + nombre + ".");
console.log("Tengo " + edad1 + " años.");
console.log("Mi estatura es " + estatura + " metros.");
console.log("Soy del género " + genero + ".");
console.log("Estoy casado: " + casado + ".");

//Funciones:
function saludar(nombre: string): void {
        console.log("Hola, " + nombre + "!");
    }
saludar("Juan");

//Clases:
    class Persona {
        nombre: string;
        edad: number;
        genero: string;
        constructor(nombre: string, edad: number, genero: string) {
        this.nombre = nombre;
        this.edad = edad;
        this.genero = genero;
        }
        presentarse(): void {
        console.log("Hola, soy " + this.nombre + ".");
        }
    }
    let juan = new Persona("Juan", 30, "M");
    juan.presentarse();
    

File: Java/0-introduccion.java
//Imprimir un mensaje:
Imprimir un mensaje:
public class HolaMundo {
    public static void main(String[] args) {
        System.out.println("Hola, mundo!");
    }
}

//Variables
public class Variables {
    public static void main(String[] args) {
        int edad = 30;
        double estatura = 1.75;
        char genero = 'M';
        boolean casado = false;
        String nombre = "Juan";
        System.out.println("Mi nombre es " + nombre + ".");
        System.out.println("Tengo " + edad + " años.");
        System.out.println("Mi estatura es " + estatura + " metros.");
        System.out.println("Soy del género " + genero + ".");
        System.out.println("Estoy casado: " + casado + ".");
    }
}

//Operaciones matemáticas:
public class Operaciones {
    public static void main(String[] args) {
        int a = 5;
        int b = 3;
        int suma = a + b;
        int resta = a - b;
        int producto = a * b;
        double cociente = (double) a / b;
        System.out.println("Suma: " + suma);
        System.out.println("Resta: " + resta);
        System.out.println("Producto: " + producto);
        System.out.println("Cociente: " + cociente);
    }
}

//Condicionales:
public class Condicionales {
    public static void main(String[] args) {
        int calificacion = 75;
        if (calificacion >= 70) {
        System.out.println("Aprobado");
        } else {
        System.out.println("Reprobado");
        }
    }
}

//Bucles:
public class Bucles {
    public static void main(String[] args) {
        for (int i = 0; i < 5; i++) {
        System.out.println(i);
        }
    }
}
public class Bucles {
    public static void main(String[] args) {
        int i = 0;
        while (i < 5) {
        System.out.println(i);
        i++;
        }
    }
}
